### ISSUE TITLE
Build instructions needed (doesn't build out of the box)

### ISSUE THREAD

**Issue by snej** [telehash/telehash-objc#17]
I'm trying to build this but not having any luck — keeps failing due to not finding the sodium headers:

```
In file included from /Code/Others/Cocoa/telehash/common/THLine.m:29:
/Code/Others/Cocoa/telehash/common/THCipherSet3a.h:10:10: fatal error: 'sodium.h' file not found
#include <sodium.h>
```

Looks like libsodium has to be built first; so I cd'd into it and did the usual "./autogen.sh && ./configure && make" thing. This build successfully, but that doesn't fix the Xcode error.

The telehash target has a header path `$(PROJECT_DIR)/libsodium/build/include`, but there's no `build` subdirectory in `libsodium`.

Any idea what to do next?

**Comment by lukedoolittle** [comment]
I agree that would be nice as I haven't gotten it to build. I am a bit of a novice with XCode so I thought it was just me. If you add a user defined header to `(solutiondir)libsodium/src/libsodium/include`, and also drop a copy of `version.h` and `export.h` into that include directory you'll get a little further but I still get stuck later in the process.

**Comment by quartzjer** [comment]
This is v2 of telehash from earlier this year, the current v3 is being worked on by @dwaite as a different approach but not ready to replace this repo yet :)

**Comment by donpark** [comment]
if you're using OS X, you can use Homebrew to install libsodium. Just

```
brew install libsodium
```

will install include files and libs at:

```
/usr/local/include/sodium
/usr/local/lib
```

**Comment by snej** [comment]
That doesn't help for iOS, though :(

**Comment by donpark** [comment]
for iOS, I just build it from source. checkout 'libsodium', build x86 version (for simulator) as documented then use `dist-build/ios.sh` to build iOS version in folder named `libsodium-ios` which will look like this.

![screenshot_190](https://cloud.githubusercontent.com/assets/127594/6156511/c2b470f4-b1ef-11e4-81e4-89c355de457f.png)
